src/models/meme.ts

```diff
@@ -0,0 +1,20 @@
+class MemeResource {
+  id: string;
+  title: string;
+  description: string;
+  link: string;
+  nsfw: boolean;
+  type: string;
+  created_at: Date;
+
+  constructor(imgurResult: JSON) {
+    this.id = imgurResult["id"];
+    this.title = imgurResult["title"];
+    this.description = imgurResult["description"];
+    this.link = imgurResult["link"];
+    this.nsfw = imgurResult["nsfw"];
+    this.type = imgurResult["type"];
```

Comment:
I wonder if you could use the destruction operator here 🤔

Reply:
Hm... 🤔 We had used something similar before, but it wasn't working 100% as intended so I changed it to the "normal" boilerplate-ish way. It was something like `constructor(public {name}: PersonDto = {})`. 

There is an interesting discussion on GitHub about this: https://github.com/Microsoft/TypeScript/issues/5326

So I will look into it tomorrow. :)

Reply:
I've read some discussions about it. But I did not fully understand it, it was a wee bit "hacky" or a way to complicated. So for now, I'd like to keep the whole constructor with it's boilerplate-ish code. It's readable and in a way a boring but solid standard approach. ;)